### DISCUSSION TITLE
How would you be able to extract the link embedded into the tweet?

### DISCUSSION THREAD

**Discussion by jonathanlin0** [tweepy/tweepy#1515] [Questions] [answered]
Like if I wanted to get the link of a Wired article that the official Wired Twitter account tweeted out of an article, how would I do that?

**Comment by Harmon758** [comment, marked as answer]
If you have the `Status`/[Tweet object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/tweet), you can use its `entities` attribute to retrieve the [entities object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#entitiesobject) that contains a `urls` field with [URL objects](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#urls) representing URLs included in the text of a Tweet.

**Reply by r0sky** [reply]
I couldn't manage to get urls from tweets shared via Socialflow. Wired, CNN, DW etc. uses Socialflow to manage their twitter accounts. When I try to get url from entities/urls or text fields, I get the link of the tweet itself instead of the embedded url. Is there any way to get thess embedded links? Example tweet ID: 1413913680757108737

**Reply by Harmon758** [reply]
You need to use extended mode for that Tweet, as it's longer than 140 characters, e.g.:
```pycon
>>> status = api.get_status(1413913680757108737, tweet_mode="extended")
>>> status.entities["urls"]
[{'url': 'https://t.co/sXUi3XmtkZ', 'expanded_url': 'https://wired.trib.al/xWowN5e', 'display_url': 'wired.trib.al/xWowN5e', 'indices': [189, 212]}]
```
See [the documentation on extended Tweets](https://tweepy.readthedocs.io/en/latest/extended_tweets.html).
